test/extended/networking/network_segmentation_endpointslice_mirror.go

```diff
@@ -0,0 +1,321 @@
+					name:     nadName,
+					topology: "layer3",
+					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
+					role:     "primary",
+				},
+				false,
+			),
+			Entry(
+				"L2 dualstack primary UDN, host-networked pods",
+				networkAttachmentConfigParams{
+					name:     nadName,
+					topology: "layer2",
+					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
+					role:     "primary",
+				},
+				true,
+			),
+			Entry(
+				"L3 dualstack primary UDN, host-networked pods",
+				networkAttachmentConfigParams{
+					name:     nadName,
+					topology: "layer3",
+					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
+					role:     "primary",
+				},
+				true,
+			),
+		)
+
+		DescribeTable(
+			"does not mirror EndpointSlices in namespaces not using user defined primary networks",
+			func(
+				netConfigParams networkAttachmentConfigParams,
+			) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+					context.Background(),
+					generateNAD(netConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deploying the backend pods")
+				replicas := 3
+				for i := 0; i < replicas; i++ {
+					runUDNPod(cs, f.Namespace.Name,
+						*podConfig(fmt.Sprintf("backend-%d", i), func(cfg *podConfiguration) {
+							cfg.namespace = f.Namespace.Name
+							// Add the net-attach annotation for secondary networks
+							if netConfig.role == "secondary" {
+								cfg.attachments = []nadapi.NetworkSelectionElement{{Name: netConfig.name}}
+							}
+						}),
+						func(pod *corev1.Pod) {
+							if pod.Labels == nil {
+								pod.Labels = map[string]string{}
+							}
+							pod.Labels["app"] = "test"
+
+						})
+				}
+
+				By("creating the service")
+				svc := e2eservice.CreateServiceSpec("test-service", "", false, map[string]string{"app": "test"})
+				familyPolicy := corev1.IPFamilyPolicyPreferDualStack
+				svc.Spec.IPFamilyPolicy = &familyPolicy
+				_, err = cs.CoreV1().Services(f.Namespace.Name).Create(context.Background(), svc, metav1.CreateOptions{})
+				framework.ExpectNoError(err, "Failed creating service %v", err)
+
+				By("asserting the mirrored EndpointSlice does not exist")
+				Eventually(func() error {
+					esList, err := cs.DiscoveryV1().EndpointSlices(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", "k8s.ovn.org/service-name", svc.Name)})
+					if err != nil {
+						return err
+					}
+
+					if len(esList.Items) != 0 {
+						return fmt.Errorf("expected no mirrored EndpointSlice, got: %d", len(esList.Items))
+					}
+					return nil
+				}, 2*time.Minute, 6*time.Second).ShouldNot(HaveOccurred())
```

Comment:
nit: positive logic reads better IMHO.

```suggestion
				}, 2*time.Minute, 6*time.Second).Should(Succeed())
```